plugin/src/main/java/com/spotify/plugin/dockerfile/MavenPomAuthSupplier.java

```diff
@@ -0,0 +1,70 @@
+/*-
+ * -\-\-
+ * Dockerfile Maven Plugin
+ * --
+ * Copyright (C) 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.plugin.dockerfile;
+
+import com.spotify.docker.client.auth.RegistryAuthSupplier;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.RegistryAuth;
+import com.spotify.docker.client.messages.RegistryConfigs;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MavenPomAuthSupplier implements RegistryAuthSupplier {
+
+  private AbstractDockerMojo abstractDockerMojo;
+
+  public MavenPomAuthSupplier(AbstractDockerMojo abstractDockerMojo) {
+    this.abstractDockerMojo = abstractDockerMojo;
```

Comment:
would make more sense to me if you passed the username and password fields into the constructor and stored them as fields of the class, rather than the mojo instance.

Reply:
👍